noir-projects/noir-contracts/contracts/test_contract/src/test.nr

```diff
@@ -77,3 +82,89 @@ fn get_shared_mutable_num_slots() -> Field {
 
     value_change_and_delay_change_packed_len + with_hash_additional_slots
 }
+
+pub unconstrained fn setup() -> (&mut TestEnvironment, AztecAddress, AztecAddress) {
+    // Setup env, generate keys
+    let mut env = TestEnvironment::new();
+    let owner = env.create_account(1);
+    env.impersonate(owner);
+
+    // TODO(#12276): This is here because we are testing historical proofs, and we cannot get a valid block_header at 1 due to a bug in world state.
+    env.advance_block_by(1);
+
+    // Deploy contract and initialize
+    let initializer = Test::interface().initialize();
+    let test_contract = env.deploy_self("Test").with_private_initializer(initializer);
+    let contract_address = test_contract.to_address();
+
+    env.advance_block_by(1);
+
+    (&mut env, contract_address, owner)
+}
+
+// These tests need to be here instead of a saner place like aztec-nr/aztec/src/history/contract_inclusion/test.nr
+// because we are unable to define nor deploy a contract from that location and currently do not have plans to change that.
+
+// In these tests we deploy a contract and can prove whether it has been deployed / initialized at specific blocks.
+#[test]
+unconstrained fn contract_historical_proofs_happy_path() {
+    let (env, contract_address) = setup();
+
+    let context = env.private();
+    let header_at_block_before_deployment = context.get_block_header_at(2);
```

Comment:
I know the header name makes it pretty clear but anyway would assign smt. like CONTRACT_DEPLOYED_AT, I would check it matches in the setup when deploying and then use the value here.